packages/components/src/mobile/keyboard-aware-flat-list/use-scroll-to-text-input.native.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+const DEFAULT_FONT_SIZE = 16;
+
+/** @typedef {import('@wordpress/element').RefObject} RefObject */
+/** @typedef {import('react-native-reanimated').SharedValue} SharedValue */
+/**
+ * Hook to scroll to the currently focused TextInput
+ * depending on where the caret is placed taking into
+ * account the Keyboard and the Header.
+ *
+ * @param {number}      extraScrollHeight      Extra space to not overlap the content.
+ * @param {number}      keyboardOffset         Keyboard space offset.
+ * @param {boolean}     scrollEnabled          Whether the scroll is enabled or not.
+ * @param {RefObject}   scrollViewMeasurements ScrollView Layout measurements.
+ * @param {RefObject}   scrollViewRef          ScrollView reference.
+ * @param {SharedValue} scrollViewYOffset      Current offset position of the ScrollView.
+ * @return {Function[]} Function to scroll to the current TextInput's offset.
+ */
+export default function useScrollToTextInput(
+	extraScrollHeight,
+	keyboardOffset,
+	scrollEnabled,
+	scrollViewMeasurements,
+	scrollViewRef,
+	scrollViewYOffset
+) {
+	const { top, bottom } = useSafeAreaInsets();
+	const insets = top + bottom;
+
+	const scrollToTextInputOffset = useCallback(
+		( caret, textInputOffset ) => {
+			const { caretHeight = DEFAULT_FONT_SIZE } = caret ?? {};
+
+			if (
+				! scrollViewRef.current ||
+				! scrollEnabled ||
+				! scrollViewMeasurements.current
+			) {
+				return;
+			}
+			const currentScrollViewYOffset = Math.max(
+				0,
+				scrollViewYOffset.value
+			);
+
+			// Scroll up.
+			if ( textInputOffset < currentScrollViewYOffset ) {
+				scrollViewRef.current.scrollTo( {
+					y: textInputOffset,
+					animated: true,
+				} );
+				return;
+			}
+
+			const availableScreenSpace = Math.abs(
+				Math.floor(
+					scrollViewMeasurements.current.height -
+						( keyboardOffset + extraScrollHeight + caretHeight )
+				)
+			);
+			const maxOffset = Math.floor(
+				currentScrollViewYOffset + availableScreenSpace
+			);
+
+			const isAtTheTop =
+				textInputOffset < scrollViewMeasurements.current.y + insets;
```

Comment:
Good question!

They don't share the same values:

- `scrollViewMeasurements` has the layout measures of the ScrollView in the window, so for this case, the `y` value is the distance of the ScrollView from the top of the screen/view.
- `scrollViewYOffset` has the value of the current scroll's Y offset of the ScrollView, so whenever we scroll that value gets updated to reflect the current scroll position offset.

Reply:
For example, if the device is in landscape mode, the scroll is at the very top and focuses on the title of the post, without this check, it would try to scroll down when it shouldn't since the title is visible.

Reply:
This would happen in the demo app, since the editor implementation and the native header are being added differently than it is done in the main apps. We had issues like this when we worked on the Drag & Drop feature.